### PR TITLE
Add an option to disable the WMS/WMTS cache

### PR DESCRIPTION
**Pull request by mmomtchev**
## What does this PR do?

Add an option to disable the WMS/WMTS cache

For very dynamic data or for environments without access to a file system

## Tasklist

 - [x] Add test case(s)
 - [ ] Add documentation
 - [ ] Review
 - [ ] Adjust for comments
 - [ ] All CI builds and checks have passed

### CODE REVIEW

File: autotest/gdrivers/wms.py [hunk 1]
```diff
@@ -371,6 +371,15 @@ def test_wms_8():
     gdal.SetConfigOption("GDAL_DEFAULT_WMS_CACHE_PATH", None)
     assert cs == 0, 'cs != 0'
 
+    # Test with GDAL_DEFAULT_WMS_CACHE_TYPE
+    # Now, we should not read from the cache anymore
+    gdal.SetConfigOption("GDAL_DEFAULT_WMS_CACHE_TYPE", "none")
```
Comment (rouault) [inline autotest/gdrivers/wms.py:376]: ``
with gdaltest.config_option("GDAL_DEFAULT_WMS_CACHE_TYPE", "none"):
     ...
`` is better now

**Review by rouault — COMMENTED** [review]
On reflection, I'm wondering if a GDAL_ENABLE_WMS_CACHE=YES/NO config option wouldn't be better rather than hacking the wms cache type

**Comment by mmomtchev** [comment]
Equivalent to the presence/absence of a `Cache` XML element?

**Comment by rouault** [comment]
> Equivalent to the presence/absence of a `Cache` XML element?

I'd say GDAL_ENABLE_WMS_CACHE=NO would overload an implicit or explicit Cache XML element

File: gdal/frmts/wms/gdalwmsdataset.cpp [hunk 2]
```diff
@@ -245,8 +245,9 @@ CPLErr GDALWMSDataset::Initialize(CPLXMLNode *config, char **l_papszOpenOptions)
     }
 
     if (ret == CE_None) {
+        const char *pszEnableCache = CPLGetConfigOption("GDAL_ENABLE_WMS_CACHE", "YES");
         CPLXMLNode *cache_node = CPLGetXMLNode(config, "Cache");
-        if (cache_node != nullptr) {
+        if (cache_node != nullptr && !EQUAL(pszEnableCache, "NO")) {
```
Comment (rouault) [inline gdal/frmts/wms/gdalwmsdataset.cpp:250]:
```suggestion
        if (cache_node != nullptr && CPLTestBool(pszEnableCache)) {
```

**Pull request merged**